crates/jxl-grid/src/shared_subgrid.rs

```diff
@@ -149,7 +149,7 @@ impl<'g, V> SharedSubgrid<'g, V> {
     /// Split the grid horizontally at an index.
     ///
     /// # Panics
-    /// Panics if `x > self.width()`.
+    /// Panics if `x >= self.width()`.
```

Comment:
What's unexpected is that it panics when `x == self.width()`; I intended the method to yield zero-sized grid in that case (just like `<[T]>::split_at()`). Maybe I broke something when refactoring `get_ptr`.

Reply:
I found that `MutableSubgrid` is written correctly and only `SharedSubgrid` is buggy. I'm going to open a new pull request with tests, please take a look at it.